src/include/storage/garbage_collector.h

```diff
@@ -24,32 +24,19 @@ class GarbageCollector {
   /**
    * Constructor for the Garbage Collector that requires a pointer to the TransactionManager. This is necessary for the
    * GC to invoke the TM's function for handing off the completed transactions queue.
-   * @param timestamp_manager source of timestamps in the system
    * @param deferred_action_manager pointer to deferred action manager of the system
    * @param txn_manager pointer to the TransactionManager
-   * @param observer the access observer attached to this GC. The GC reports every record gc-ed to the observer if
-   *                 it is not null. The observer can then gain insight invoke other components to perform actions.
-   *                 The observer's function implementation needs to be lightweight because it is called on the GC
-   *                 thread.
    */
   // TODO(Tianyu): Eventually the GC will be re-written to be purely on the deferred action manager. which will
   //  eliminate this perceived redundancy of taking in a transaction manager.
```

Comment:
Can we get rid of this TODO now?

Reply:
let's leave it here for now that as we still keep fake gc there to keep the return format consistent.